irohad/consensus/yac/impl/supermajority_checker_impl.cpp

```diff
@@ -23,10 +23,10 @@ namespace iroha {
     namespace yac {
 
       bool SupermajorityCheckerImpl::hasSupermajority(
-          const shared_model::interface::SignatureSetType &signatures,
+          const shared_model::interface::SignatureRangeType &signatures,
           const std::vector<std::shared_ptr<shared_model::interface::Peer>>
               &peers) const {
-        return checkSize(signatures.size(), peers.size())
+        return checkSize(boost::range_detail::range_calculate_size(signatures), peers.size())
```

Comment:
Please add `.size()` method to `Signable` class, because `range_detail` does not look like public API, and it probably has to iterate the `signatures` in linear time, what can be avoided because the size is known in `Signable`.